### ISSUE TITLE
Make OwnersClient directory blacklist configurable.

### ISSUE THREAD

**Issue by cjwagner** [kubernetes/test-infra#5156]
This blacklist specifies any directories that should not be considered when looking for OWNERS files.
It is currently [hard coded](https://github.com/kubernetes/test-infra/blob/master/prow/owners/owners.go#L43) as `[".git", "_output"]`.

Requested in #5017 cc @kargakis

**Comment by stevekuznetsov** [comment]
Should we just use `.gitignore`?

**Comment by cjwagner** [comment]
I don't think that `.gitignore` is sufficient. There may be checked in code that shouldn't be considered when looking for OWNERS files. (Maybe `vendor/`?)

**Comment by stevekuznetsov** [comment]
Good point -- I swear we had turned off `OWNERS` in `vendor` at some point, or we had to add a check that the owner was actually someone in the repo under test so the assign wouldn't fail.

**Comment by cjwagner** [comment]
When I adapted the OWNERS file support in mungegithub for Prow I made sure that we filter out users who aren't collaborators since mungegithub had problems with that.
https://github.com/kubernetes/test-infra/blob/dbee4115d7033719bd336b6fad7c8d4b63285da6/prow/repoowners/repoowners.go#L169-L179
It would still be best to skip large directories that we don't care about. The `approve` plugin is very slow compared to other plugins and avoiding walking directories like `vendor/` could improver performance quite a bit.
